src/qt/pivx/guitransactionsutils.cpp

```diff
@@ -7,77 +7,83 @@
             case WalletModel::DuplicateAddress:
-                msgParams.first = parent->translate(
+                retStr = parent->translate(
                         "Duplicate address found, can only send to each address once per send operation.");
                 break;
             case WalletModel::TransactionCreationFailed:
-                msgParams.first = parent->translate("Transaction creation failed!");
-                msgParams.second = CClientUIInterface::MSG_ERROR;
+                retStr = parent->translate("Transaction creation failed!");
+                informType = CClientUIInterface::MSG_ERROR;
                 break;
             case WalletModel::TransactionCommitFailed:
-                msgParams.first = parent->translate(
+                retStr = parent->translate(
                         "The transaction was rejected! This might happen if some of the coins in your wallet were already spent, such as if you used a copy of wallet.dat and coins were spent in the copy but not marked as spent here.");
-                msgParams.second = CClientUIInterface::MSG_ERROR;
+                informType = CClientUIInterface::MSG_ERROR;
                 break;
             case WalletModel::AnonymizeOnlyUnlocked:
                 // Unlock is only need when the coins are send
                 if (!fPrepare)
                     fAskForUnlock = true;
                 else
-                    msgParams.first = parent->translate("Error: The wallet is unlocked for staking only. Fully unlock the wallet to send the transaction.");
+                    retStr = parent->translate("Error: The wallet is unlocked for staking only. Fully unlock the wallet to send the transaction.");
                 break;
 
             case WalletModel::InsaneFee:
-                msgParams.first = parent->translate(
+                retStr = parent->translate(
                         "A fee %1 times higher than %2 per kB is considered an insanely high fee.").arg(10000).arg(
                         BitcoinUnits::formatWithUnit(walletModel->getOptionsModel()->getDisplayUnit(),
                                                      ::minRelayTxFee.GetFeePerK()));
                 break;
                 // included to prevent a compiler warning.
             case WalletModel::OK:
             default:
-                return;
+                return retStr; // No issue
         }
 
         // Unlock wallet if it wasn't fully unlocked already
         if (fAskForUnlock) {
             walletModel->requestUnlock(AskPassphraseDialog::Context::Unlock_Full, false);
             if (walletModel->getEncryptionStatus() != WalletModel::Unlocked) {
-                msgParams.first = parent->translate(
+                retStr = parent->translate(
                         "Error: The wallet was unlocked only to anonymize coins. Unlock canceled.");
             } else {
                 // Wallet unlocked
-                return;
+                return retStr;
```

Comment:
Nit: the `else` shouldn't be needed.
Also, wouldn't it be more organized to move this block inside `if(!fPrepare)` so there's nothing (except `return retStr;`) outside the swtich?

Other nit:
> The wallet was unlocked only to anonymize coins

should be changed to "unlocked for staking only".

Reply:
done